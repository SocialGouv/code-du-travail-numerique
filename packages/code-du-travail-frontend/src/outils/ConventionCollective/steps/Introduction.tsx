import { SOURCES } from "@socialgouv/cdtn-sources";
import {
  AlertWithIcon,
  Button,
  Paragraph,
  theme,
  Tile as TileUi,
} from "@socialgouv/cdtn-ui";
import Link from "next/link";
import React from "react";
import styled from "styled-components";

import {} from "../../common/type/WizardType";
import { ScreenType } from "../common/NavContext";
import { useTrackingContext } from "../common/TrackingContext";

const IntroductionStep = (): JSX.Element => {
  const { trackEvent, uuid, title } = useTrackingContext();

  return (
    <>
      <AlertWithIcon variant="secondary">
        <p>
          La convention collective est un texte conclu au niveau d’une branche
          d’activité (Ex : Transports routiers). Elle adapte les règles du code
          du travail aux situations particulières de la branche (primes, congés,
          salaires minima, préavis, prévoyance…).
        </p>
        <Paragraph noMargin fontWeight="700">
          Vous pouvez retrouver le nom de votre convention collective sur votre
          bulletin de paie ou sur votre contrat de travail.
        </Paragraph>
      </AlertWithIcon>
      <Flex>
        <Link
          href={`/${SOURCES.TOOLS}/convention-collective#${ScreenType.agreement}`}
          passHref
        >
          <Tile
            onClick={() =>
              trackEvent("cc_search_type_of_users", " click_p1", title, uuid)
            }
          >
            <Paragraph noMarginfontWeight="700">
              Je connais
              <br /> ma convention collective
            </Paragraph>
            <ButtonWrapper>
              <Button variant="link" as="div">
                Je la saisis
              </Button>
            </ButtonWrapper>
          </Tile>
        </Link>
        <Link
          href={`/${SOURCES.TOOLS}/convention-collective#${ScreenType.enterprise}`}
          passHref
        >
          <Tile
            onClick={() =>
              trackEvent("cc_search_type_of_users", " click_p2", title, uuid)
            }
          >
            <Paragraph noMargin fontWeight="700">
              Je ne connais <br /> pas ma convention collective
            </Paragraph>
            <ButtonWrapper>
              <Button variant="link" as="div">
                Je la recherche
              </Button>
            </ButtonWrapper>
          </Tile>
        </Link>
      </Flex>
    </>
  );
};

export { IntroductionStep };

const Flex = styled.div`
  display: flex;
  padding-top: ${theme.spacings.medium};
  justify-content: space-around;
  @media (max-width: ${theme.breakpoints.mobile}) {
    flex-direction: column;
    align-items: center;
  }
`;

const Tile = styled(TileUi)`
  flex: 0 1 auto;
  width: 28rem;
  @media (max-width: ${theme.breakpoints.mobile}) {
    & + & {
      margin-top: ${theme.spacings.medium};
    }
  }
`;

const ButtonWrapper = styled.div`
  padding-top: ${theme.spacings.medium};
`;
