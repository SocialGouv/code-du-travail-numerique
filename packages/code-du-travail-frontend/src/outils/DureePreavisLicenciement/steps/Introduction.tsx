import Link from "next/link";
import React from "react";

const StepIntro = (): JSX.Element => (
  <>
    <p>
      Le préavis est une période accordée au salarié entre la notification du
      licenciement et son départ effectif de l’entreprise (fin du contrat).
    </p>

    <p>
      Afin de réaliser cette simulation, vous aurez besoin d’informations
      concernant la situation de l’employé concerné, comme sa convention
      collective, sa catégorie ou son coefficient. La plupart de ces
      informations se trouvent sur le bulletin de salaire ou le contrat de
      travail.
    </p>

    <p>
      Vous pouvez trouver plus d’informations sur le préavis de licenciement sur{" "}
      <Link href={`/fiche-service-public/preavis-de-licenciement`}>
        <a>cette fiche</a>
      </Link>
      .
    </p>
  </>
);

export { StepIntro };
