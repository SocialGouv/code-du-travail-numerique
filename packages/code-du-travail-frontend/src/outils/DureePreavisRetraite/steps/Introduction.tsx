import { Alert } from "@socialgouv/cdtn-ui";
import React from "react";

const IntroductionStep = (): JSX.Element => (
  <>
    <p>
      Ce simulateur est un outil qui permet de calculer la durée de préavis à
      respecter en cas de départ ou de mise à la retraite, telle qu&apos;elle
      est prévue par la loi et la convention collective applicable.
    </p>
    <p>
      En effet, en cas de départ ou de mise à la retraite, le salarié ne peut
      pas quitter l’entreprise immédiatement. Il doit y rester durant une
      certaine période, qu’on appelle préavis ou parfois délai congé.
    </p>
    <p>
      Afin de réaliser cette simulation, vous aurez besoin d’informations
      concernant la situation du salarié, comme sa convention collective, sa
      catégorie ou son coefficient. La plupart de ces informations se trouvent
      sur le bulletin de salaire ou le contrat de travail. Prévoyez X à X
      minutes pour cette simulation.
    </p>
    <Alert>
      Attention&nbsp;: Le résultat affiché correspond à la durée légale ou
      conventionnelle, en fonction des informations renseignées. Toutefois, une
      durée plus favorable peut être prévue par un accord collectif
      d’entreprise, le contrat de travail, un usage ou encore une convention
      collective de branche (si non renseignée dans le simulateur). Dans ce cas,
      c’est cette durée plus favorable qui s’applique au salarié.
    </Alert>
    <p>Cliquez sur commencer pour démarrer la simulation.</p>
  </>
);

export { IntroductionStep };
