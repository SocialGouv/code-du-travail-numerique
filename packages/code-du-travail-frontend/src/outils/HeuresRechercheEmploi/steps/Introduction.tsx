import Link from "next/link";
import React from "react";

const StepIntro = (): JSX.Element => (
  <>
    <p>
      Ce simulateur permet de calculer le nombre d’heures d’absence autorisée
      pendant la période de préavis (de licenciement, de démission ou de rupture
      de la période d’essai), pour rechercher un emploi, si la convention
      collective le prévoit.
    </p>
    <p>
      Des informations disponibles sur le bulletin de salaire pourraient être
      utiles pour réaliser cette simulation.
    </p>
    <p>
      Vous pouvez trouver plus d’informations sur les heures d’absence autorisée
      pour rechercher un emploi pendant le préavis sur{" "}
      <Link
        href={`/contribution/le-salarie-peut-il-sabsenter-pour-rechercher-un-emploi-pendant-son-preavis`}
      >
        cette fiche
      </Link>
      .
    </p>
  </>
);

export { StepIntro };
