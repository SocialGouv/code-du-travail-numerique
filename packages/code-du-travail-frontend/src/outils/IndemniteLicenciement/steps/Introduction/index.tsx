import React from "react";

const StepIntro = (): JSX.Element => (
  <>
    <p>
      Ce simulateur permet d’estimer le montant de l’indemnité légale de
      licenciement d’un salarié.
    </p>
    <p>
      Vous aurez besoin d’informations telles que les dates d’entrée et de
      sortie de l’entreprise, la date de notification du licenciement et le
      montant des derniers salaires.
    </p>
  </>
);

export default StepIntro;
