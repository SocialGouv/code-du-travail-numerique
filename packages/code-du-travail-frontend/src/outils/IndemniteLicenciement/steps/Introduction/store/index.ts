export { default as createIntroductionStore } from "./store";
export * from "./types";
