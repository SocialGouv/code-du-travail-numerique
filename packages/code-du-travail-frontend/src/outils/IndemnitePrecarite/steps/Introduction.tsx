import React from "react";

const StepIntro = (): JSX.Element => (
  <>
    <p>
      Ce simulateur est un outil qui permet d’estimer le montant de l’indemnité
      de fin de contrat ou de fin de mission (dite “prime de précarité”) d’un
      salarié en CDD ou en contrat d’intérim.
    </p>
    <p>
      Vous pourrez simuler simplement le montant de la prime de précarité
      définie par le Code du travail avec la prise en compte des particularités
      présentes dans certaines branches d’activité.
    </p>
    <p>
      Afin de pouvoir remplir les renseignements demandés, munissez-vous des
      informations relatives au contrat de travail (CDD ou contrat de travail
      temporaire (contrat d’intérim), derniers bulletins de salaire, …).
    </p>
  </>
);

export { StepIntro };
