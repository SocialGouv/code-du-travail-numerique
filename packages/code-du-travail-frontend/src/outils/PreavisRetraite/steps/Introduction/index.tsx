import React from "react";

const StepIntro = (): JSX.Element => (
  <>
    <p>
      Ce simulateur vous permet de calculer la durée de préavis à respecter en
      cas de départ ou de mise à la retraite.
    </p>
    <p>
      Des informations disponibles sur le bulletin de salaire pourraient être
      utiles pour réaliser cette simulation.
    </p>
  </>
);

export default StepIntro;
