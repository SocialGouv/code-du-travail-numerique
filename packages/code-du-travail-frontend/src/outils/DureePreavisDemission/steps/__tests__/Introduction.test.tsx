import { renderForm } from "../../../../../test/renderForm";
import { StepIntro } from "../Introduction";

describe("<StepIntro />", () => {
  it("should render", () => {
    const { container } = renderForm(StepIntro);
    expect(container).toMatchSnapshot();
  });
});
