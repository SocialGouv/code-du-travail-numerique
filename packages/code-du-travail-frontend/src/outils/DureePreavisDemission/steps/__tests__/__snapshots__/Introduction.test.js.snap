// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepIntro /> should render 1`] = `
<div>
  <p>
    En cas de démission, le salarié ne peut pas quitter l’entreprise immédiatement après avoir informé l’employeur de ses intentions. Il doit rester dans l’entreprise durant une certaine période, qu’on appelle préavis ou parfois délai congé.
  </p>
  <p>
    Les règles encadrant le préavis (droits et obligations, durées, cas de dispense, absences autorisées pour chercher un emploi durant le préavis) sont notamment définies par le code du travail et la branche.
  </p>
  <p>
    Le présent outil vous permet de connaitre la durée du préavis prévue par la convention collective en matière de démission.
  </p>
</div>
`;
